File: 10_flutter_and_native_samples/android/app/src/main/kotlin/com/example/flutter_and_native_samples/NativeViewFactory.kt
package com.example.flutter_and_native_samples

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/** Creates the native Android view requested by Flutter's AndroidView widget. */
class NativeViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return NativeLabelView(context)
    }
}

File: 10_flutter_and_native_samples/ios/Runner/NativeViewFactory.swift
import Flutter
import UIKit

/// Creates the native iOS view requested by Flutter's UiKitView widget.
final class NativeViewFactory: NSObject, FlutterPlatformViewFactory {
  func create(
    withFrame frame: CGRect,
    viewIdentifier viewId: Int64,
    arguments args: Any?
  ) -> FlutterPlatformView {
    NativeLabelView(frame: frame)
  }
}
